### PR TITLE
Bugfix/clear form button

### LINKED ISSUE #143: Clear form does not seem to work
#### Describe the bug

clear form does not work

#### To Reproduce

Steps to reproduce the behaviour:

1. Go to form
2. fill form
3. click clear form
4. does not work

#### Expected behavior

clear form

### PR DESCRIPTION
**Pull request by saulipurhonen**
### Description

Fixes issue where form doesn't clear when continuing draft.
Added check for form submit that form is valid.

### Related issues

Closes #143 

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

### Changes Made

- Reset form with form schema as parameter. This was with default values before and therefore wouldn't clear when editing draft
- Check that form is valid when submitting form. Without this current draft item would be deleted.

### Testing

- [x] E2E Tests

### CODE REVIEW

File: src/components/NewDraftWizard/WizardForms/WizardFillObjectDetailsForm.js [hunk 1]
```diff
@@ -153,6 +153,7 @@ const FormContent = ({ resolver, formSchema, onSubmit, objectType, folderId }: F
     if (currentDraftId || draftObject.accessionId) {
       const response = await draftAPIService.patchFromJSON(objectType, currentDraftId, cleanedValues)
       if (response.ok) {
+        console.log(response)
```
Comment (blankdots) [inline src/components/NewDraftWizard/WizardForms/WizardFillObjectDetailsForm.js:156]: forgotten console log ?

Comment (saulipurhonen) [reply src/components/NewDraftWizard/WizardForms/WizardFillObjectDetailsForm.js]: Removed and rebased.

**Review by blankdots — APPROVED** [review]

**Review by lilachic — APPROVED** [review]
Looks good to me. Just a note that it clears a form but it still alerts asking to cancel/save, when trying to change to another object type.

**Comment by blankdots** [comment]
@lilachic maybe that is another bug e.g. if the form is empty changing from it it should not raise that alert, or @saulipurhonen should we treat as part of this issue. I remember this was solved some time ago in #77 ?

**Comment by saulipurhonen** [comment]
@lilachic, @blankdots that indeed is a bug and is issued in #127. I'll update the issue with more details regarding this behaviour.

**Comment by hannyle** [comment]
looks good to me.

**Review by hannyle — APPROVED** [review]

**Comment by blankdots** [comment]
merging as is and the identified bug will be fix in #127

**Pull request merged**